### DISCUSSION TITLE
Provide a launch file.

### DISCUSSION THREAD

**Discussion by ryleymcc** [OpenVSLAM-Community/openvslam_ros#49] [Q&A] [answered]
I am using realsense t265 and d435i but I'm not sure how to create the launch file.
I placed created a directory called setup and put the .yaml and .fbow files in it.
This is what i have now but it gives me an error saying bad file.
```
<launch>
    <node name="openvslam" pkg="openvslam_ros" type="run_slam" args="-v $(find openvslam_ros)/setup/orb_vocab.fbow -c $(find openvslam_ros)/setup/config.yaml" />
</launch>
```
Here is terminal output 
```
$ roslaunch openvslam_ros t265.launch 
... logging to /home/ryley/.ros/log/d64f0962-30be-11ec-a6df-773cae5178b4/roslaunch-ryley-GL65-9SD-147000.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://ryley-GL65-9SD:35323/

SUMMARY
========

PARAMETERS
 * /rosdistro: noetic
 * /rosversion: 1.15.13

NODES
  /
    openvslam (openvslam_ros/run_slam)

auto-starting new master
process[master]: started with pid [147008]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to d64f0962-30be-11ec-a6df-773cae5178b4
process[rosout-1]: started with pid [147018]
started core service [/rosout]
process[openvslam-2]: started with pid [147021]
bad file
[openvslam-2] process has died [pid 147021, exit code 1, cmd /home/ryley/catkin_ws/devel/lib/openvslam_ros/run_slam -v /home/ryley/catkin_ws/src/openvslam_ros/setup/orb_vocab.fbow -c /home/ryley/catkin_ws/src/openvslam_ros/setup/config.yaml __name:=openvslam __log:=/home/ryley/.ros/log/d64f0962-30be-11ec-a6df-773cae5178b4/openvslam-2.log].
log file: /home/ryley/.ros/log/d64f0962-30be-11ec-a6df-773cae5178b4/openvslam-2*.log
```

**Comment by ymd-stella** [comment, marked as answer]
Make sure that `$(find openvslam_ros)/setup/config.yaml` exists.

**Reply by ryleymcc** [reply]
well that's embarrassing.  I think i just needed to sleep but yea that's the issue. Thanks

**Comment by ryleymcc** [comment]
If anyone needs a starting point in the future, here is a basic launch file to get you started using the t265 realsense camera.
make sure you put the yaml and vocab files are in the right place.
```
<launch>
    <remap from="/t265/fisheye1/image_raw" to="/camera/image_raw"/>
    <include file="$(find realsense2_camera)/launch/rs_d400_and_t265.launch" >
        <arg name="enable_fisheye" value="true" />
    </include>
    <node name="openvslam" pkg="openvslam_ros" type="run_slam" args="-v $(find openvslam_ros)/src/setup/orb_vocab.fbow -c $(find openvslam_ros)/src/setup/fisheye.yaml" />
</launch>
```

**Reply by mgrallos** [reply]
> If anyone needs a starting point in the future, here is a basic launch file to get you started using the t265 realsense camera. make sure you put the yaml and vocab files are in the right place.
> 
> ```
> <launch>
>     <remap from="/t265/fisheye1/image_raw" to="/camera/image_raw"/>
>     <include file="$(find realsense2_camera)/launch/rs_d400_and_t265.launch" >
>         <arg name="enable_fisheye" value="true" />
>     </include>
>     <node name="openvslam" pkg="openvslam_ros" type="run_slam" args="-v $(find openvslam_ros)/src/setup/orb_vocab.fbow -c $(find openvslam_ros)/src/setup/fisheye.yaml" />
> </launch>
> ```

the config file is the parameters of the camera used correct?

**Reply by ymd-stella** [reply]
Don't write another question to a resolved question.
